src/components/AuthForm.js

```diff
@@ -0,0 +1,119 @@
+import React from "react";
+import Button from "react-bootstrap/Button";
+import {
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+} from "firebase/auth";
+import { auth } from "../firebase";
+
+class AuthForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      emailInputValue: "",
+      passwordInputValue: "",
+      isNewUser: true,
+      errorCode: "",
+      errorMessage: "",
+    };
+  }
+
+  handleInputChange = (event) => {
+    this.setState({ [event.target.name]: event.target.value });
+  };
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    if (!this.state.emailInputValue || !this.state.passwordInputValue) return;
+
+    const closeAuthForm = () => {
+      // Reset auth form state
+      this.setState({
+        emailInputValue: "",
+        passwordInputValue: "",
+        isNewUser: true,
+        errorCode: "",
+        errorMessage: "",
+      });
```

Comment:
here we could use the default state object we talked about during your post-mortem